### PR TITLE
Fixed a bug where the Publication Date in the News sitemap would be wrong

### LINKED ISSUE #227: Wrong publication date in news sitemap
The _news:publication_date_ is set to UTC time, however, a wrong time offset is applied (see image below).

When using UTC, the time offset should be +00:00. Instead the time offset corresponding to the local time is outputted (+02:00) leading to a wrong publication time.

As a result, the article appears older in the search results, as if it had been published 2 hours before its actual publication time.

image

Related: https://github.com/Yoast/wpseo-news/issues/112

### PR DESCRIPTION
**Pull request by Dieterrr**
## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the Publication Date in the News sitemap would be wrong when a hardcoded timezone was set (for example UTC + 2). 

## Relevant technical choices:

* Class-sitemap-item now returns the UTC time by default in the news sitemap, if `post_date_gmt` is for the post. If not, it defaults to 'post_date' and returns the local time + timezone modifier. 
* Removed fallbacks relying on `post_modified` data, as this data is not available there and this always threw an error. 
* Removed code from 'class-sitemap-item' that attempted to guess the timezone, and replaced this with a conversion into a usable hard coded time string.   

## Test instructions

This PR can be tested by following these steps:

**How to reproduce:** 
* Make sure the Yoast test helper is activated with 'Disable the XML sitemaps cache.' checked. 
* Now go to `Settings->General`, set timezone to **"Amsterdam"**.
* Create a new post, put the local UTC time in the title (just to make it easier to read). In this example let's say it's '14:19'. If you want to double check the current utc time,  type "current utc time" in google, to make sure you are comparing the news sitemap time with the correct time. 
* Go to `Yoast settings > News SEO > View news sitemap`. 
* The news sitemap should read:  "14:19:05+02:00", Which is incorrect. 

**How to test:** 
* Now checkout this branch and repeat the steps. 
* The time in the news sitemap should now be the correct UTC time without a modifier. For example. If the timezone setting for your website is on Amsterdam and your post is published at 11:00, the time in the sitemap should be 9:00. 

* timezone **"Amsterdam"**:
The news sitemap should read:  "14:19:05+00:00", Which is correct. 

* Timezone **"UTC+2"**.
The news sitemap should read:  "14:19:05+00:00", Which is correct. 

* Finally, The fallback for having no `post_date_gmt` set for your post is to my knowledge not testable, as this value seems to be always set to a valid value, even if you delete it from the database. 

Fixes #227.

This is only 1/2 of the fix of #227. This fixes the wrong time in the news sitemap, but not in the publication_date metadata. This will be fixed in wordpress-seo in a separate issue.

### CODE REVIEW

File: classes/class-sitemap-item.php [hunk 1]
```diff
@@ -240,18 +231,14 @@ private function get_publication_date( $item ) {
 	 *
 	 * @param string $item_date Date to parse.
 	 *
+	 * @param string $time_zone Timezone to parse.
```
[the remark this answers is not in the document]
Comment (Dieterrr) [reply classes/class-sitemap-item.php]: Yes, but wouldn't this call it's `__toString()` method by default?

File: classes/class-sitemap-timezone.php [hunk 2]
```diff
@@ -35,29 +35,30 @@ public function wp_get_timezone_string() {
...
-
-		// Attempt to guess the timezone string from the UTC offset.
-		$timezone = timezone_name_from_abbr( '', $utc_offset );
+		// Format the UTC offset to a string readable by DateTimeZone.
+		$offset_float = floatval( $utc_offset );
 
-		if ( false !== $timezone ) {
-			return $timezone;
+		if ( $utc_offset < 0 ) {
+			$offset_float *= -1;
```
[the remark this answers is not in the document]
Comment (Dieterrr) [reply classes/class-sitemap-timezone.php]: Instead of a comment I made it more readable by using abs()

**Comment by Xyfi** [comment]
CR :+1:

**Comment by IreneStr** [comment]
I've tested this branch, and it looked good. However, I could not reproduce the original issue completely, which is why I asked @Xyfi for another acceptance test.

**Comment by Xyfi** [comment]
So I've looked into the issue. And it seems that the prettified output of the sitemap differs from the actual xml source.

I posted the article named `http://local.wordpress.test/newer-news-by-news-the-news-noose` at `2018-09-12T08:38:41+00:00`(GMT).

## Without this PR it produces the following sitemaps:

**"Pretty" output: (Correct)**
screen shot 2018-09-12 at 10 51 53

**Source: (Incorrect)**
screen shot 2018-09-12 at 10 53 32

This is incorrect, because:
`new Date( "2018-09-12T08:38:41+02:00" ).toGMTString(); // => Wed, 12 Sep 2018 06:38:41 GMT`

## With this PR it produces the following sitemaps:

**"Pretty" output: (Correct)**
screen shot 2018-09-12 at 10 59 17

**Source: (Correct)**
screen shot 2018-09-12 at 11 00 01

So I'm merging this PR and leaving this here for future reference.

**Pull request merged**